### PR TITLE
Content & media url endpoints

### PR DESCRIPTION
**Pull request by Migaroez**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
- New dedicated endpoints for retrieving URLs by key (e.g. for displaying picked item URLs in the content/media picker)
- Async misusage cleanup in factories

### Testing
- Create and publish some documents (preferably some in multiple cultures, don't forget to assign domains)
- Upload some temporary files and create some media from it (see example payload below)
- The following endpoints should return the relevant url info for all found documents/media by the keys you supplied `/umbraco/amanagement/api/v1/document/urls` `/umbraco/amanagement/api/v1/media/urls`

### Example payloads
Media with image cropper backing field
```
{
    "mediaType": {
        "id": "cc07b313-0843-4aa8-bbda-871c8da728c8"
    },
    "values": [
        {
            "culture": null,
            "segment": null,
            "alias": "umbracoFile",
            "value": {"src":"5344f6ae-d305-47fc-955f-3a7778b6c336"}
        }
    ],
    "variants": [
        {
            "culture": null,
            "segment": null,
            "name": "test media"
        }
    ]
}
```
Media with a generic file backing field
{
    "mediaType": {
        "id": "4c52d8ab-54e6-40cd-999c-7a5f24903e4d"
    },
    "values": [
        {
            "culture": null,
            "segment": null,
            "alias": "umbracoFile",
            "value": "5344f6ae-d305-47fc-955f-3a7778b6c336"
        }
    ],
    "variants": [
        {
            "culture": null,
            "segment": null,
            "name": "test file"
        }
    ]
}

### CODE REVIEW

File: src/Umbraco.Cms.Api.Management/Controllers/Document/DocumentUrlController.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        IContentService contentService,
+        IDocumentUrlFactory documentUrlFactory)
+    {
+        _contentService = contentService;
+        _documentUrlFactory = documentUrlFactory;
+    }
+
+    [HttpGet]
+    [MapToApiVersion("1.0")]
+    [HttpGet("urls")]
```
Comment (Zeegaan) [inline src/Umbraco.Cms.Api.Management/Controllers/Document/DocumentUrlController.cs:28]: No need to have `[HttpGet]` twice 😁
```suggestion
    [HttpGet("urls")]
    [MapToApiVersion("1.0")]
```

File: src/Umbraco.Cms.Api.Management/ViewModels/Document/DocumentUrlInfoResourceSet.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Umbraco.Cms.Api.Management.ViewModels.Document;
+
+public sealed class DocumentUrlInfoResourceSet
```
Comment (Zeegaan) [inline src/Umbraco.Cms.Api.Management/ViewModels/Document/DocumentUrlInfoResourceSet.cs:3]: I'm unsure about naming here, as our usual naming is either suffixed with `ViewModel` or `ResponseModel` 🤔 
Maybe we should discuss this naming with the team, just to align 😁

File: src/Umbraco.Cms.Api.Management/ViewModels/Media/MediaUrlInfoResourceSet.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Umbraco.Cms.Api.Management.ViewModels.Media;
+
+public sealed class MediaUrlInfoResourceSet
```
Comment (Zeegaan) [inline src/Umbraco.Cms.Api.Management/ViewModels/Media/MediaUrlInfoResourceSet.cs:3]: Same as the comment in `DocumentUrlInfoResourceSet`

File: src/Umbraco.Cms.Api.Management/Controllers/Media/MediaUrlController.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        IMediaUrlFactory mediaUrlFactory)
+    {
+        _mediaService = mediaService;
+        _mediaUrlFactory = mediaUrlFactory;
+    }
+
+    [HttpGet]
+    [MapToApiVersion("1.0")]
+    [HttpGet("urls")]
+    [ProducesResponseType(typeof(IEnumerable<MediaUrlInfoResourceSet>), StatusCodes.Status200OK)]
```
Comment (Zeegaan) [inline src/Umbraco.Cms.Api.Management/Controllers/Media/MediaUrlController.cs:28]:
```suggestion
    [HttpGet("urls")]
    [MapToApiVersion("1.0")]
    [ProducesResponseType(typeof(IEnumerable<MediaUrlInfoResourceSet>), StatusCodes.Status200OK)]
```

**Review by Zeegaan — CHANGES_REQUESTED** [review]
Had a few comments before getting to testing 😁

**Review by Zeegaan — APPROVED** [review]
@Migaroez I changed the naming on the resourcesets, please confirm these are the correct names, then we are ready to merge 😁

**Comment by Migaroez** [comment]
I am happy with the rename. Merging

**Pull request merged**